src/hdmf/spec/spec.py

```diff
@@ -32,7 +32,7 @@ class DtypeHelper:
         'long': ["int64", "long"],
         'utf': ["text", "utf", "utf8", "utf-8"],
         'ascii': ["ascii", "bytes"],
-        'bool': ["bool"],
+        'bool': ["bool", "bool_"],
```

Comment:
```suggestion
        'bool': ["bool"],
```
This change is not necessary unless we want to add a new alias "bool_" for "bool" for dtype specs in the spec yaml.

Reply:
Ah, ok, that makes the purpose of `primary_dtype_synonyms` more clear, thanks for clarifying.

I added `bool_` here in order to have it added to `validator.__allowable` (see https://github.com/agencyenterprise/hdmf/blob/enh/np_bool_in_validator/src/hdmf/validate/validator.py#L18-L40), since that seemed like the most minimally invasive way to resolve the issue, but I didn't know the side-effects of updating `primary_dtype_synonyms`. Removing `bool_` here will make the new test fail, so we need to find another solution.

I originally wanted to do something like this: https://github.com/hdmf-dev/hdmf/pull/203 but that would either require a more significant change to how the validator checks types in `check_type` (currently by matching against lists in `__allowable`), or make `bool` a unique case.

I've found another implementation here which I think is a good middle-ground and relatively consistent with the existing implementation. Let me know what you think.

For the longer term - do you think this solution is acceptable, or should we add an issue to try and bring the type checking in the validator in line with the method used for type checking in docval (https://github.com/hdmf-dev/hdmf/pull/203)?